### PR TITLE
Hackweek/android q

### LINKED ISSUE #1274: Hack week: Android Q
My hack week project is ensuring WCAndroid works with SDK 29 (Q), which will involve the following:

- [x] Temporarily upgrade `compileSdkVersion`, `targetSdkVersion`, and `buildToolsVersion` to 29
- [x] Fix the errors and warnings that pop up when building the app after updating to SDK 29
- [x] Upgrade login library to SDK 29
- [x] Upgrade to AndroidX preferences
- [x] Verify the app works with the new Android Q gestural navigation
- [x] Verify the app isn’t affected by the Android Q behavior changes
- [x] Verify the app continues to run on API 21 devices
- [x] Experiment with edge-to-edge UI (most likely won’t suit our app)
- [x] Revert back to SDK 28 (we can make the actual change to SDK 29 once Q is out of beta)

Note that I’ve decided not to do anything regarding the new Android Q dark theme since that would need design work ahead of time.

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #1274 - as stated in the issue, this PR is to prepare us for Android Q. Only one reviewer is necessary.

Note that we're _not_ updating to SDK 29 in this PR since Android Q is still in beta. 

- Temporarily upgraded `compileSdkVersion`, `targetSdkVersion`, and `buildToolsVersion` to 29
- Fixed the errors and warnings that popped up when building the app after updating to SDK 29
- Fixed a number of unrelated warnings (apologies, I should've done that in a separate PR)
- Upgraded to AndroidX preferences
- Verified the app works with the new Android Q gestural navigation
- Verified the app isn’t affected by the Android Q behavior changes
- Verified the app continues to run on API 21 devices
- Experimented with edge-to-edge UI and decided it didn't suit our app
- Reverted back to SDK 28 and verified everything still works

I decided not to do anything regarding the new Android Q dark theme since that would need design work ahead of time.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -95,7 +95,7 @@ class MainActivity : AppCompatActivity(),
         }
     }
 
-    @Inject lateinit var fragmentInjector: DispatchingAndroidInjector<androidx.fragment.app.Fragment>
+    @Inject lateinit var fragmentInjector: DispatchingAndroidInjector<Fragment>
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt]: I like that this `<androidx.fragment.app.Fragment>`  was removed here but I see a couple of places in the app [here](https://github.com/woocommerce/woocommerce-android/blob/36f2070c4bd528b225a1be5bc1f9421ad7686240/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt#L38) and [here](https://github.com/woocommerce/woocommerce-android/blob/36f2070c4bd528b225a1be5bc1f9421ad7686240/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/AppSettingsActivity.kt#L112) where it is still being used. Is the plan to fix this as part of #1276? If so, I'm good to merge these changes 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListPresenter.kt [hunk 2]
```diff
@@ -180,7 +178,7 @@ class NotifsListPresenter @Inject constructor(
     }
 
     @Suppress("unused")
-    @Subscribe(threadMode = MAIN)
+    @Subscribe(threadMode = ThreadMode.MAIN)
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/notifications/NotifsListPresenter.kt]: np: There are some places in the app where this is not yet fixed. Is this part of the #1276 fix? If so, I'm good to merge 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/AddOrderTrackingProviderListAdapter.kt [hunk 3]
```diff
@@ -161,11 +161,11 @@ class AddOrderTrackingProviderListAdapter(
     ) {
         override fun getContentItemsTotal() = list.size
 
-        override fun getItemViewHolder(view: View): androidx.recyclerview.widget.RecyclerView.ViewHolder {
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/AddOrderTrackingProviderListAdapter.kt]: np: There are a lot of places in the app where we still use `androidx.recyclerview.widget.RecyclerView` in the code. Is the plan to fix those in #1276 ?

**Comment by nbradbury** [comment]
Thanks for the review @anitaa1990! Regarding the code cleanup, I started to do it in this PR but stopped after realizing how many changes needed to be made. I'd like to complete the changes when I tackle #1276 rather than add more to this PR.

**Pull request merged**